frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +75,56 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and kind
+ */
+export const useModelFinder = () => {
+  const models: K8sKind[] = _.values(
+    useSelector(({ k8s }) => k8s.getIn(['RESOURCES', 'models'])).toJS(),
+  );
+  const groupVersionMap: DiscoveryResources['groupVersionMap'] = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'groupToVersionMap']),
+  );
+
+  const findModel = (group: string, kind: string) => {
+    const groupObject = _.find(groupVersionMap, (item) => item.name === group);
+    const { preferredVersion, versions } = groupObject || {};
+    let model = undefined;
+    if (preferredVersion) {
+      // Find a model for the CRD that uses this preferred version
+      model = _.find(
+        models,
+        (item) =>
+          item.apiGroup === group &&
+          item.apiVersion === preferredVersion &&
+          (item.kind === kind || item.plural === kind),
+      );
```

Comment:
You should be able to get a model by reference without iterating over all models. (See `modelFor`.)

Reply:
Yup this would be a much better approach.